src/pl/senla/hotel/controller/ControllerRoomReservation.java

```diff
@@ -0,0 +1,7 @@
+package pl.senla.hotel.controller;
+
+import pl.senla.hotel.entity.RoomReservation;
+
+public interface ControllerRoomReservation extends ControllerCRUDALL<RoomReservation>,
```

Comment:
Let's not create these intermediate interfaces, let's use multiple implementations in classes. Because now you have 4 classes in this package, there are 12 interfaces. It is very problematic to read and understand the code. I had to download the repository to the PC

Reply:
which is optimal quantity of interfaces for 4 classes?

Reply:
it's hard to telд, to be honest :D
Look, your code should always be simple, easy to read and cover the task at hand. Now you are trying to cover all the following developments of your application by creating abstract interfaces and inheriting them. Right now it is unnecessary and redundant, the code is too cumbersome and difficult to understand. Right now, you can make separate interfaces for each class, and it will be ok even if they have duplicate methods, but with different return types. But, I'm not trying to wean you from optimization, what you tried to do well, but it turned out superfluous. Let's do this, for now, make one common interface with СRUD operations and 4 additional ones that will extend it with the methods necessary for classes. And only then, when the project grows, you can either expand or narrow the abstraction

Reply:
Similar approach to repositories also

Reply:
I'try to explain for which reason I created intermediate interfaces.

For example Repository for entity Guest: 
before: **RepositoryGuestImpl implements RepositoryGuest**.
after: **RepositoryGuestImpl implements Repository<Guest>, RepositoryAdditional**.
We exclude intermediate interface RepositoryGuest which extended earlier 2 above mentioned interfaces.
So we will reduce total quantity of interfaces in package from 12 at least to 5-7.

But in service layer in this case to save incupsulation I must create 2 instances which work with methods from RepositoryGuestImpl.
before:
    private final RepositoryGuest guestRepository = new RepositoryGuestCollection();
after:
    private final RepositoryCRUDALL<Guest> guestRepositoryCRUDALL = new RepositoryGuestCollection();
    private final RepositoryGuestAdditional guestRepositoryAdditional = new RepositoryGuestCollection();
It's not a problem. I can refactor this in 1 hour.

I can go in such ways:
1. I'll delete intermadiate interfaces and add additional entities to upper layers (as it is shown above) if I understand your proposal correctly;
2. I'll ave present quantity of interfaces but divide them to different subpackages (repository.guest, repository.reservation, etc.). It will make structure easier readable.
3. Perhaps you could offer another more efficient way on base of your experience.

Reply:
ready